CMakeLists.txt

```diff
@@ -448,31 +448,22 @@ endif ()
 
 if (NOT DISABLE_PYTHON_BINDINGS)
   set(Python_ADDITIONAL_VERSIONS 3)
-  find_package(PythonInterp)
-  if (NOT PYTHONINTERP_FOUND)
+  find_package(Python COMPONENTS Interpreter Development)
+  if (NOT Python_FOUND)
     message(STATUS "Skipping Python bindings: Python interpreter not found")
   endif ()
 
   if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/bindings/python/3rdparty/pybind11/CMakeLists.txt")
     message(WARNING "Skipping Python bindings: pybind11 submodule not available")
-    set(PYTHONINTERP_FOUND false)
+    set(Python_FOUND false)
   endif ()
 
-  if (${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR} VERSION_LESS 3.5)
+  if (${Python_VERSION} VERSION_LESS 3.5)
     message(WARNING "Skipping Python bindings: Python 3.5 or greater required")
-    set(PYTHONINTERP_FOUND false)
+    set(Python_FOUND false)
   endif ()
```

Comment:
I wonder if we can't do the following changes instead of messing with `Python_FOUND`:

- Check for `bindings/python/3rdparty/pybind11/CMakeLists.txt` before this block and force `DISABLE_PYTHON_BINDINGS` to `OFF` to skip the entire `find_package` block.
- Use `find_package(Python 3.5 COMPONENTS Interpreter Development)` instead of checking the version later by hand.

Reply:
Yeah, this is a problem in other places too. I wanted to talk to Robin about our `FindRequiredPackage` macro before I did that, but he agrees that we should just use the built-in functionality instead. It gives good error messages on failures.

Reply:
Fixed both of those recommendations in that last fixup.